Cannot use SDL_image, SDL_ttf with CMake

### Description / Steps to reproduce the issue

1. Install SDL image, TTF, etc: `pacboy --noconfirm -S --needed SDL_image:p SDL_ttf:p cmake:p toolchain:p`
1. Basic main: `echo 'int main() { return 0; }' > main.c`
1. Create this `CMakeLists.txt` file:

```cmake
cmake_minimum_required(VERSION 3.31)
project(test1)

include(FindPkgConfig)

pkg_check_modules(SDL_IMAGE SDL_image REQUIRED IMPORTED_TARGET GLOBAL)
pkg_check_modules(SDL_TTF SDL_ttf REQUIRED IMPORTED_TARGET GLOBAL)

add_executable(main main.c)
target_link_libraries(main PRIVATE PkgConfig::SDL_IMAGE PkgConfig::SDL_TTF)
```

4. Run `cmake -B build .`

I have a very old piece of software that needs to use SDL_image and SDL_ttf version 1 as well as SDL 1.2.

### Expected behavior

Should not have a problem generating the build files.

### Actual behavior

```
-- Building for: Ninja
-- The C compiler identification is Clang 20.1.8
-- The CXX compiler identification is Clang 20.1.8
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/msys64/clang64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/msys64/clang64/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: C:/msys64/clang64/bin/pkg-config.exe (found version "2.5.1")
-- Checking for module 'SDL_image'
--   Found SDL_image, version 1.2.12
-- Checking for module 'SDL_ttf'
--   Found SDL_ttf, version 2.0.11
-- Configuring done (1.7s)
CMake Error in CMakeLists.txt:
  Imported target "PkgConfig::SDL_IMAGE" includes non-existent path

    "/clang64/include/SDL"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

### Verification

- [x] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-26100

### MINGW environments affected

- [x] MINGW64
- [x] MINGW32
- [x] UCRT64
- [x] CLANG64
- [ ] CLANGARM64

### Are you willing to submit a PR?

Yes

--- a/mingw-w64-pkgconf/PKGBUILD
+++ b/mingw-w64-pkgconf/PKGBUILD
@@ -5,7 +5,7 @@ pkgbase=mingw-w64-${_realname}
 pkgname="${MINGW_PACKAGE_PREFIX}-${_realname}"
 epoch=1
 pkgver=2.5.1
-pkgrel=1
+pkgrel=2
 pkgdesc='pkg-config compatible utility which does not depend on glib'
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clangarm64')
@@ -21,11 +21,17 @@ conflicts=("${MINGW_PACKAGE_PREFIX}-pkg-config")
 replaces=("${MINGW_PACKAGE_PREFIX}-pkg-config")
 makedepends=("${MINGW_PACKAGE_PREFIX}-meson" "${MINGW_PACKAGE_PREFIX}-cc")
 groups=("${MINGW_PACKAGE_PREFIX}-toolchain")
-source=("https://github.com/pkgconf/pkgconf/archive/pkgconf-$pkgver/$_realname-$pkgver.tar.gz")
-sha256sums=('79721badcad1987dead9c3609eb4877ab9b58821c06bdacb824f2c8897c11f2a')
+source=("https://github.com/pkgconf/pkgconf/archive/pkgconf-$pkgver/$_realname-$pkgver.tar.gz"
+        "pkgconf-issue-535-minimal-backport.patch")
+sha256sums=('79721badcad1987dead9c3609eb4877ab9b58821c06bdacb824f2c8897c11f2a'
+            '77d01b1ed6a1d7d16a6ac45567fb8534ac27657766acad138c259d72c84b6b22')
 
 prepare() {
   mv "${_realname}-${_realname}-${pkgver}" "${_realname}-${pkgver}"
+
+  cd "${_realname}-${pkgver}"
+  # https://github.com/pkgconf/pkgconf/issues/535
+  patch -p1 -i "${srcdir}"/pkgconf-issue-535-minimal-backport.patch
 }
 
 build() {

--- a/mingw-w64-pkgconf/pkgconf-issue-535-minimal-backport.patch
+++ b/mingw-w64-pkgconf/pkgconf-issue-535-minimal-backport.patch
@@ -1,0 +1,65 @@
+diff --git a/libpkgconf/pkg.c b/libpkgconf/pkg.c
+index 491c0de..c80269b 100644
+--- a/libpkgconf/pkg.c
++++ b/libpkgconf/pkg.c
+@@ -65,6 +65,20 @@ str_has_suffix(const char *str, const char *suffix)
+	return !strncasecmp(str + str_len - suf_len, suffix, suf_len);
+ }
+ 
++static inline char *
++path_sep_rchr(char *buf)
++{
++	char *pathbuf = strrchr(buf, PKG_DIR_SEP_S);
++#ifdef _WIN32
++	char *altpathbuf = strrchr(buf, '/');
++
++	if (altpathbuf != NULL && (pathbuf == NULL || altpathbuf > pathbuf))
++		pathbuf = altpathbuf;
++#endif
++
++	return pathbuf;
++}
++
+ static char *
+ pkg_get_parent_dir(pkgconf_pkg_t *pkg)
+ {
+@@ -130,9 +144,7 @@ pkg_get_parent_dir(pkgconf_pkg_t *pkg)
+	}
+ #endif
+ 
+-	pathbuf = strrchr(buf, PKG_DIR_SEP_S);
+-	if (pathbuf == NULL)
+-		pathbuf = strrchr(buf, '/');
++	pathbuf = path_sep_rchr(buf);
+	if (pathbuf != NULL)
+		pathbuf[0] = '\0';
+ 
+@@ -283,15 +295,11 @@ determine_prefix(const pkgconf_pkg_t *pkg, char *buf, size_t buflen)
+	pkgconf_strlcpy(buf, pkg->filename, buflen);
+	pkgconf_path_relocate(buf, buflen);
+ 
+-	pathiter = strrchr(buf, PKG_DIR_SEP_S);
+-	if (pathiter == NULL)
+-		pathiter = strrchr(buf, '/');
++	pathiter = path_sep_rchr(buf);
+	if (pathiter != NULL)
+		pathiter[0] = '\0';
+ 
+-	pathiter = strrchr(buf, PKG_DIR_SEP_S);
+-	if (pathiter == NULL)
+-		pathiter = strrchr(buf, '/');
++	pathiter = path_sep_rchr(buf);
+	if (pathiter == NULL)
+		return NULL;
+ 
+@@ -301,9 +309,7 @@ determine_prefix(const pkgconf_pkg_t *pkg, char *buf, size_t buflen)
+ 
+	/* okay, work backwards and do it again. */
+	pathiter[0] = '\0';
+-	pathiter = strrchr(buf, PKG_DIR_SEP_S);
+-	if (pathiter == NULL)
+-		pathiter = strrchr(buf, '/');
++	pathiter = path_sep_rchr(buf);
+	if (pathiter == NULL)
+		return NULL;
+ 

